File: getcorona.pl
#!/usr/bin/env perl
#
# File name: getcorona.pl
# Date:      2020/03/21 14:00
# Author:    Nigel Houghton <wutang@warpten.net>
# $Id$
# Copyright: (c) Nigel Houghton 2020
#
#############################################################################
#

use strict;
use warnings;

use JSON qw( );
use LWP::UserAgent;

my $ua = LWP::UserAgent->new;
$ua->agent("Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_4) AppleWebKit/600.7.12 (KHTML, like Gecko) Version/8.0.7 Safari/600.7.12");
 
# Create a request
my $req = HTTP::Request->new(GET => 'https://corona-stats.online/US?format=json');
$req->content_type('application/x-www-form-urlencoded');
$req->content('query=libwww-perl&mode=dist');
 
# Pass request to the user agent and get a response back
my $res = $ua->request($req);
my $json_text; 
# Check the outcome of the response
if ($res->is_success) {
  $json_text = $res->content;
}
else {
  print $res->status_line, "\n";
  exit;
}

# Now get our JSON sortedout
my $json = JSON->new;
my $data = $json->decode($json_text);

# US Population as of April 6, 2020
# my $uspopulation = 329480988;
my $uspopulation = $data->{data}[0]{population};
my $mortalityrate = sprintf("%.2f", $data->{data}[0]{deaths}/$data->{data}[0]{confirmed} * 100);
my $recoveryrate = sprintf("%.2f", $data->{data}[0]{recovered}/$data->{data}[0]{confirmed} * 100);
my $percentinfected = sprintf("%.2f", $data->{data}[0]{confirmed}/$uspopulation * 100);

print $data->{data}[0]{country} . ":" . "\n"
  . "\t" . "Cases: " . $data->{data}[0]{confirmed} . "\n"
  . "\t(" . $percentinfected . "% of total population)" . "\n"
  . "\tDeaths: " . $data->{data}[0]{deaths}
  . "\tRecovered: " . $data->{data}[0]{recovered} . "\n"
  . "\tMortality Rate: " . $mortalityrate . "%"
  . "\tRecovery Rate: " . $recoveryrate . "%\n";

__END__

=head1 getcorona.pl

=head1 DESCRIPTION

=head1 USAGE
  Just run it.

=head1 AUTHOR
  Nigel Houghton <wutang@warpten.net>

=head1 COPYRIGHT
  (c) Nigel Houghton 2020

File: getcorona.py
#!/opt/local/bin/python
"""
 COVID-19 Stats for the US

 Copyright: Nigel Houghton <wutang@warpten.net>
 $Id$

"""

import urllib.request
import json
import requests
import time

url = "https://corona-stats.online/US"
querystring = {"format":"json"}
response = requests.request("GET", url, headers=headers, params=querystring)
data = response.json()

# If you want to see the JSON data, uncomment the next line
# print(json.dumps(data, indent = 4, sort_keys = True))

# Format numbers to use comma separators
def place_value(number):
    return("{:,}".format(number))
# Pretty print text strings to line things up a bit
def pretty_print(str):
    return("{:15}".format(str))

for info in data['data']:
    deaths = info['deaths']
    cases = info['cases']
    population = info['population']
    todaycases = info['todayCases']
    recovered = info['recovered']
    datatime = info['updated']
    datatime = time.strftime("%A, %B %d %Y %r", time.localtime(int(datatime/1000)))
    deathrate = str(round((deaths/cases)*100, 2)) + "%"
    recoveryrate = str(round((recovered/cases)*100, 2)) + "%"
    percentinfected = str(round((cases/population)*100, 2)) + "%"

print("As of", datatime)
print(pretty_print("US Population:"),place_value(population))
print(pretty_print("Cases:"),place_value(cases))
print(pretty_print("Deaths:"),place_value(deaths))
print(pretty_print("New Cases:"),place_value(todaycases))
print(pretty_print("Death rate:"),deathrate)
print(pretty_print("Recovery rate:"),recoveryrate)
print(pretty_print("Infection rate:"),percentinfected)

